probe/appclient/app_client.go

```diff
@@ -311,7 +311,10 @@ func (c *appClient) pipeConnection(id string, pipe xfer.Pipe) (bool, error) {
 	defer c.closeConn(id)
 
 	_, remote := pipe.Ends()
-	return false, pipe.CopyToWebsocket(remote, conn)
+	if err := pipe.CopyToWebsocket(remote, conn); err != nil && !xfer.IsExpectedWSCloseError(err) {
+		return false, err
+	}
+	return false, nil
```

Comment:
This is a change in behaviour. Previously any error at this stage would be handled by the backoff logic. Now only unexpected errors cause backoff (and are logged); expected errors cause an immediate retry.

@paulbellamy @fons is this bad?

We could return `true, nil` instead, which would cause the backoff loop to be aborted. Would that be better? What conditions is the backoff supposed to handle?